jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3404,9 +3404,13 @@ private static MethodHandle validateArgumentsOfWhileLoop(MethodHandle predHandle
 		Class<?> bodyReturnType = bodyType.returnType;
 		
 		/* The signature of the body handle must be either (V, A...)V or (A...)void */
-		if ((void.class != bodyReturnType) && (bodyReturnType != bodyType.arguments[0])) {
+		if ((void.class != bodyReturnType) 
+				&& ((0 == bodyType.arguments.length) 
+						|| (bodyReturnType != bodyType.arguments[0]))) {
+			String argList = (0 == bodyType.arguments.length) ? "void" : bodyType.arguments[0].getName();
```

Comment:
We generally try to avoid single line if statements as they make it difficult to set breakpoints that only fire if the if is true.  Code like:
```
String argList = "void";
if (bodyType.arguments.length > 0) {
    argList = bodyType.arguments[0].getName();
}
```
is as clear and allows setting a breakpoint on the code in the if.